### PR TITLE
Remove triage workflow in favor of github built-in automation

### PR DESCRIPTION
**Pull request by r0qs**
Since this workflow now has a built-in replacement :)
See: https://github.blog/changelog/2023-03-09-github-issues-projects-march-9th-update/#robot-automatically-add-and-archive-project-items

### CODE REVIEW

**Review by cameel — APPROVED** [review]

File: .github/workflows/triage.yml [hunk 1]
```diff
@@ -1,127 +0,0 @@
-name: Add new issues to triage column
-
-on:
-  issues:
-    types:
-      - opened
-
-env:
-  GH_TOKEN: ${{ secrets.PROJECT_BOARD_AUTOMATION }}
```
Comment (cameel) [inline .github/workflows/triage.yml:9]: I guess we can now remove this secret? This seems to be the only workflow using it.

Comment (r0qs) [reply .github/workflows/triage.yml:9]: yes :)

Comment (cameel) [reply .github/workflows/triage.yml:9]: Have you already enabled the automation in the workflow?

Comment (r0qs) [reply .github/workflows/triage.yml:9]: yes I have. You can check it here: https://github.com/orgs/ethereum/projects/27/workflows

**Comment by cameel** [comment]
`PROJECT_BOARD_AUTOMATION` secret removed.

**Pull request merged**